db/eventlog.c

```diff
@@ -547,34 +534,40 @@ static void eventlog_add_int(cson_object *obj, const struct reqlogger *logger)
     eventlog_path(obj, logger);
 }
 
+static inline void add_to_fingerprints(const struct reqlogger *logger)
+{
+    bool isSql = logger->event_type && (strcmp(logger->event_type, "sql") == 0);
+    bool isSqlErr = logger->error && logger->stmt;
+
+    if ((isSql || isSqlErr) && !hash_find(seen_sql, logger->fingerprint)) {
+        eventlog_add_newsql(logger);
+    }
+}
+
 void eventlog_add(const struct reqlogger *logger)
 {
-    Pthread_mutex_lock(&eventlog_lk);
     if (eventlog == NULL || !eventlog_enabled) {
```

Comment:
Cursory check should be fine without lock. We do another check below under lock.